### PR TITLE
Implement Pgrange overlaps_with

### PR DESCRIPTION
**Pull request by vijairamg**
Implementing `anyrange && anyrange -> boolean` as part of #4092

### CODE REVIEW

File: diesel/src/pg/expression/helper_types.rs [hunk 1]
```diff
@@ -59,6 +59,10 @@ pub type RangeContains<Lhs, Rhs> = Grouped<
     >,
 >;
 
+/// The return type of [`lhs.overlaps_with(rhs)`](super::expression_methods::PgRangeExpressionMethods::overlaps_with)
+/// for range expressions
+#[cfg(feature = "postgres_backend")]
+pub type RangeOverlaps<Lhs, Rhs> = OverlapsWith<Lhs, Rhs>;
```
Comment (weiznich) [inline diesel/src/pg/expression/helper_types.rs:65]: I believe it's not necessary to have this type def as it matches the typedef of the array operators in line 41. We can just adjust the doc comment there and mention that this is also the return type of `PgRangeExperssionMethods::overlaps_with(rhs)`.

**Review by weiznich — CHANGES_REQUESTED** [review]
Thanks for submitting this PR and sorry for taking some time for doing a review.

The change looks mostly good beside the minor note about the new type def. It can be just replaced by the existing one.

File: diesel/src/pg/expression/helper_types.rs (hunk 1 above)
Comment (vijairamg) [reply diesel/src/pg/expression/helper_types.rs]: I have removed the new type declaration and added the comment.

**Review by weiznich — APPROVED** [review]

**Comment by weiznich** [comment]
Thanks for the update. It looks good now.

**Pull request merged**